scripts/velero.sh

```diff
@@ -14,54 +14,156 @@ CLUSTER_NAME=$1
+  AZURE_LOCATION=$(yq e .azure.location $PARAMS_YAML)
+  AZURE_APP_NAME=$(yq e .azure.app-name $PARAMS_YAML)
+  AZURE_SUBSCRIPTION_ID=$(yq e .azure.subscription-id $PARAMS_YAML)
+  AZURE_TENANT_ID=$(yq e .azure.tenant-id $PARAMS_YAML)
+  AZURE_CLIENT_ID=$(yq e .azure.client-id $PARAMS_YAML)
+  AZURE_CLIENT_SECRET=$(yq e .azure.client-secret $PARAMS_YAML)
+  AZURE_CLOUD_NAME=$(yq e .azure.environment $PARAMS_YAML)
+
+  AZURE_BACKUP_RESOURCE_GROUP=$CLUSTER_NAME
+  AZURE_STORAGE_ACCOUNT_ID=tkglabbackups
+  BLOB_CONTAINER=velero-$CLUSTER_NAME
+
+  # validate existance of storage account
+  if [ -z "`az storage account list -o tsv --query "[?name=='$AZURE_STORAGE_ACCOUNT_ID']"`" ]; then
+    # storage account doesn't exist, so create it
+    echo "Creating storage account for velero: $AZURE_STORAGE_ACCOUNT_ID"
+    az storage account create \
+      --name $AZURE_STORAGE_ACCOUNT_ID \
+      --resource-group $AZURE_BACKUP_RESOURCE_GROUP \
+      --sku Standard_GRS \
+      --encryption-services blob \
+      --https-only true \
+      --kind BlobStorage \
+      --access-tier Hot
+  else
+    echo "Storage account for velero already exists: $AZURE_STORAGE_ACCOUNT_ID"
+  fi
+
+  # validate existance of container
+  if [ -z "`az storage container list --account-name $AZURE_STORAGE_ACCOUNT_ID -o tsv --query "[?name=='$BLOB_CONTAINER']"`" ]; then
+    # blob container doesn't exist, so create it
+    echo "Creating blob container for velero: $BLOB_CONTAINER"
+    az storage container create -n $BLOB_CONTAINER --public-access off --account-name $AZURE_STORAGE_ACCOUNT_ID
+  else
+    echo "Blob container for velero already exists: $BLOB_CONTAINER"
+  fi
+
+  cat << EOF  > generated/$CLUSTER_NAME/velero/credentials-velero
+AZURE_SUBSCRIPTION_ID=${AZURE_SUBSCRIPTION_ID}
+AZURE_TENANT_ID=${AZURE_TENANT_ID}
+AZURE_CLIENT_ID=${AZURE_CLIENT_ID}
+AZURE_CLIENT_SECRET=${AZURE_CLIENT_SECRET}
+AZURE_RESOURCE_GROUP=${AZURE_BACKUP_RESOURCE_GROUP}
+AZURE_CLOUD_NAME=${AZURE_CLOUD_NAME}
+EOF
 
-  echo "Using IAM Profile for S3 Access"
   velero install \
       --image=$VELERO_IMAGE_REPO/tkg/velero/velero:$VELERO_VERSION \
-      --provider aws \
-      --plugins $VELERO_IMAGE_REPO/tkg/velero/velero-plugin-for-aws:$VELERO_PLUGIN_FOR_AWS_VERSION \
-      --bucket $VELERO_BUCKET \
-      --backup-location-config region=$VELERO_REGION \
-      --snapshot-location-config region=$VELERO_REGION \
-      --no-secret
+      --provider azure \
+      --plugins $VELERO_IMAGE_REPO/tkg/velero/velero-plugin-for-microsoft-azure:$VELERO_PLUGIN_FOR_AZURE_VERSION \
+      --bucket $BLOB_CONTAINER \
+      --secret-file generated/$CLUSTER_NAME/velero/credentials-velero \
+      --backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,storageAccount=$AZURE_STORAGE_ACCOUNT_ID \
+
+elif [ "$IAAS" = "aws" ]; then
+
+
+  echo aws
+  AWS_REGION=$(yq e .aws.region $PARAMS_YAML)
+
+  if [ -z "$AWS_SESSION_TOKEN" ]; then
+  # this condition is for AWS without CloudGate
```

Comment:
Why this condition is based on `AWS_SESSION_TOKEN`?

Reply:
There are two flavors for AWS.  One is if you have a cloudgate account that rotates your keys and uses an AWS_SESSION_TOKEN and thus uses a node permission for velero to access S3.  The other is non-cloud gate and that works as normal.

Reply:
Got it, I read the conditional check backwards